### ISSUE TITLE
Add an option for "new-parens" to require having no parenthesis on new

### ISSUE THREAD

**Issue by pfgithub** [eslint/eslint#10034]
#10074 

**What rule do you want to change?**

`new-parens`

**Does this change cause the rule to produce more or fewer warnings?**

No

**How will the change be implemented? (New option, new default behavior, etc.)?**

Adds an option for always requiring or always not allowing new parens

**Please provide some example code that this change will affect:**

`"new-parens": ["error", "never"]`
Examples of incorrect code for this rule:
```js
var person = new Person();
var person = new (Person)();
var thing = new Person().Thing
```
Examples of correct code for this rule:
```js
var person = new Person;
var person = new (Person);
var person = new Person("Name");
var thing = (new Person).Thing
```

`"new-parens": ["error", "always"]`
Examples of incorrect code for this rule:
```js
var person = new Person;
var person = new (Person);
var thing = (new Person).Thing;
```
Examples of correct code for this rule:
```js
var person = new Person();
var person = new (Person)();
var person = new Person("Name");
var thing = new Person().Thing;
```

**What does the rule currently do for this code?**

Has no option of setting to require no parenthesis on new with no arguments

**What will the rule do after it's changed?**

Same thing as it currently does, but also includes an option to require no parenthesis on new constructors with no arguments

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. How would the rule handle cases like this when using the `never` option?

```js
new Foo().Bar
```

**Comment by pfgithub** [comment]
> Thanks for the proposal. How would the rule handle cases like this when using the never option?
> 
> new Foo().Bar

This would be a warning/error. If --fix is applied it would surround it in parenthesis like `(new Foo).Bar`
